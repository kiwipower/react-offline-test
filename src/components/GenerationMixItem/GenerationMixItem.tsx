import React from 'react'

import { GeneratorPercentageItem } from '../../services/GenerationData'

interface GenerationMixItemProps {
    gdata: GeneratorPercentageItem;
    color: string;
}

export default function GenerationMixItem(props: GenerationMixItemProps) {
    
    const style = {
        width: props.gdata.perc + '%',
        backgroundColor: props.color,
    }

    return (
        <div className="chart-item row">
        <div className="col-sm-2 chart-item__label">
            {props.gdata.fuel} 
        </div>
        <div className="col-sm-1 chart-item__perc">
            <span>({props.gdata.perc}%)</span>
        </div>
        <div className="col-sm-9">
            <div className="progress">
                <div className="progress-bar" role="progressbar" 
                        style={style} 
                        aria-valuenow={props.gdata.perc} 
                        aria-valuemin={0} 
                        aria-valuemax={100}>
                        
                </div>
            </div>
        </div>
        
    </div>
    )
}
