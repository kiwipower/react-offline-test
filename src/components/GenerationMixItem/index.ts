import GenerationMixItem from "./GenerationMixItem";

export default GenerationMixItem; 