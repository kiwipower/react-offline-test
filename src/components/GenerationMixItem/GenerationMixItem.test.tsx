import React from 'react';
import { shallow } from 'enzyme';

import GenerationMixItem from './GenerationMixItem';

describe('<GenerationMixItem />', () => {

    const renderComponent = () => (<GenerationMixItem gdata={{
        "fuel": "gas",
        "perc": 46.5
    }} color="tomato" />);

    it('renders without error', async () => {
        const component = shallow(renderComponent());
        expect(component).toMatchSnapshot();
    });

});