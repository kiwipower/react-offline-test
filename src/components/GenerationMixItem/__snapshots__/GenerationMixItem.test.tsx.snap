// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GenerationMixItem /> renders without error 1`] = `
<div
  className="chart-item row"
>
  <div
    className="col-sm-2 chart-item__label"
  >
    gas
  </div>
  <div
    className="col-sm-1 chart-item__perc"
  >
    <span>
      (
      46.5
      %)
    </span>
  </div>
  <div
    className="col-sm-9"
  >
    <div
      className="progress"
    >
      <div
        aria-valuemax={100}
        aria-valuemin={0}
        aria-valuenow={46.5}
        className="progress-bar"
        role="progressbar"
        style={
          Object {
            "backgroundColor": "tomato",
            "width": "46.5%",
          }
        }
      />
    </div>
  </div>
</div>
`;
